--- a\minicraft\level\tile\DoorTile.java
+++ b\minicraft\level\tile\DoorTile.java
@@ -50,7 +50,7 @@
 		if (item instanceof ToolItem) {
 			ToolItem tool = (ToolItem) item;
 			if (tool.type == type.getRequiredTool()) {
-				if (player.payStamina(4 - tool.level) && tool.payDurability()) {
+				if (player.payStamina(4 - ToolItem.LEVELS.get(tool.level)) && tool.payDurability()) {
 					level.setTile(xt, yt, Tiles.get(id + 3)); // Will get the corresponding floor tile.
 					Sound.monsterHurt.play();
 					level.dropItem(xt * 16 + 8, yt * 16 + 8, Items.get(type.name() + " Door"));

