--- a\minicraft\level\tile\FlowerTile.java
+++ b\minicraft\level\tile\FlowerTile.java
@@ -15,10 +15,10 @@
 
 public class FlowerTile extends Tile {
 	private static final Sprite flowerSprite = new Sprite(3, 8, 1);
-	
+
 	protected FlowerTile(String name) {
 		super(name, (ConnectorSprite)null);
-		connectsToGrass = true;
+		connections.add(TileConnections.TileConnectionType.GRASS);
 		maySpawn = true;
 	}
 
@@ -37,16 +37,16 @@
 		}
 		return false;
 	}
-	
+
 	public void render(Screen screen, Level level, int x, int y) {
 		Tiles.get("Grass").render(screen, level, x, y);
-		
+
 		int data = level.getData(x, y);
 		int shape = (data / 16) % 2;
-		
+
 		x = x << 4;
 		y = y << 4;
-		
+
 		flowerSprite.render(screen, x + 8 * shape, y);
 		flowerSprite.render(screen, x + 8 * (shape == 0 ? 1 : 0), y + 8);
 	}
@@ -55,7 +55,7 @@
 		if (item instanceof ToolItem) {
 			ToolItem tool = (ToolItem) item;
 			if (tool.type == ToolType.Shovel) {
-				if (player.payStamina(2 - tool.level) && tool.payDurability()) {
+				if (player.payStamina(2 - ToolItem.LEVELS.get(tool.level)) && tool.payDurability()) {
 					level.setTile(x, y, Tiles.get("Grass"));
 					Sound.monsterHurt.play();
 					level.dropItem(x * 16 + 8, y * 16 + 8, Items.get("Flower"));

