--- a\minicraft\level\tile\GrassTile.java
+++ b\minicraft\level\tile\GrassTile.java
@@ -20,7 +20,7 @@
 			return tile.connectsToGrass;
 		}
 	};
-	
+
 	protected GrassTile(String name) {
 		super(name, sprite);
 		csprite.sides = csprite.sparse;
@@ -31,10 +31,10 @@
 	public boolean tick(Level level, int xt, int yt) {
 		// TODO revise this method.
 		if (random.nextInt(40) != 0) return false;
-		
+
 		int xn = xt;
 		int yn = yt;
-		
+
 		if (random.nextBoolean()) xn += random.nextInt(2) * 2 - 1;
 		else yn += random.nextInt(2) * 2 - 1;
 
@@ -54,7 +54,7 @@
 		if (item instanceof ToolItem) {
 			ToolItem tool = (ToolItem) item;
 			if (tool.type == ToolType.Shovel) {
-				if (player.payStamina(4 - tool.level) && tool.payDurability()) {
+				if (player.payStamina(4 - ToolItem.LEVELS.get(tool.level)) && tool.payDurability()) {
 					level.setTile(xt, yt, Tiles.get("Dirt"));
 					Sound.monsterHurt.play();
 					if (random.nextInt(5) == 0) { // 20% chance to drop Grass seeds
@@ -64,7 +64,7 @@
 				}
 			}
 			if (tool.type == ToolType.Hoe) {
-				if (player.payStamina(4 - tool.level) && tool.payDurability()) {
+				if (player.payStamina(4 - ToolItem.LEVELS.get(tool.level)) && tool.payDurability()) {
 					level.setTile(xt, yt, Tiles.get("Dirt"));
 					Sound.monsterHurt.play();
 					if (random.nextInt(5) != 0) { // 80% chance to drop Wheat seeds
@@ -74,7 +74,7 @@
 				}
 			}
 			if (tool.type == ToolType.Pickaxe) {
-				if (player.payStamina(4 - tool.level) && tool.payDurability()) {
+				if (player.payStamina(4 - ToolItem.LEVELS.get(tool.level)) && tool.payDurability()) {
 					level.setTile(xt, yt, Tiles.get("Path"));
 					Sound.monsterHurt.play();
 				}

