--- a\minicraft\level\tile\GrassTile.java	2021-09-08 21:34:58.000000000 +0800
+++ b\minicraft\level\tile\GrassTile.java	2022-09-13 19:28:35.683654400 +0800
@@ -54,7 +54,7 @@
 		if (item instanceof ToolItem) {
 			ToolItem tool = (ToolItem) item;
 			if (tool.type == ToolType.Shovel) {
-				if (player.payStamina(4 - tool.level) && tool.payDurability()) {
+				if (player.payStamina(4 - ToolItem.LEVELS.get(tool.level)) && tool.payDurability()) {
 					level.setTile(xt, yt, Tiles.get("Dirt"));
 					Sound.monsterHurt.play();
 					if (random.nextInt(5) == 0) { // 20% chance to drop Grass seeds
@@ -64,7 +64,7 @@
 				}
 			}
 			if (tool.type == ToolType.Hoe) {
-				if (player.payStamina(4 - tool.level) && tool.payDurability()) {
+				if (player.payStamina(4 - ToolItem.LEVELS.get(tool.level)) && tool.payDurability()) {
 					level.setTile(xt, yt, Tiles.get("Dirt"));
 					Sound.monsterHurt.play();
 					if (random.nextInt(5) != 0) { // 80% chance to drop Wheat seeds
@@ -74,7 +74,7 @@
 				}
 			}
 			if (tool.type == ToolType.Pickaxe) {
-				if (player.payStamina(4 - tool.level) && tool.payDurability()) {
+				if (player.payStamina(4 - ToolItem.LEVELS.get(tool.level)) && tool.payDurability()) {
 					level.setTile(xt, yt, Tiles.get("Path"));
 					Sound.monsterHurt.play();
 				}
