--- a\minicraft\level\tile\LavaBrickTile.java
+++ b\minicraft\level\tile\LavaBrickTile.java
@@ -13,16 +13,16 @@
 
 public class LavaBrickTile extends Tile {
 	private static Sprite sprite = new Sprite(19, 2, 2, 2, 1);
-	
+
 	protected LavaBrickTile(String name) {
 		super(name, sprite);
 	}
-	
+
 	public boolean interact(Level level, int xt, int yt, Player player, Item item, Direction attackDir) {
 		if (item instanceof ToolItem) {
 			ToolItem tool = (ToolItem) item;
 			if (tool.type == ToolType.Pickaxe) {
-				if (player.payStamina(4 - tool.level) && tool.payDurability()) {
+				if (player.payStamina(4 - ToolItem.LEVELS.get(tool.level)) && tool.payDurability()) {
 					level.setTile(xt, yt, Tiles.get("Lava"));
 					Sound.monsterHurt.play();
 					return true;

