--- a\minicraft\level\tile\OreTile.java
+++ b\minicraft\level\tile\OreTile.java
@@ -21,26 +21,26 @@
 public class OreTile extends Tile {
 	private Sprite sprite;
 	private OreType type;
-	
+
 	public enum OreType {
         Iron (Items.get("Iron Ore"), 0),
 		Lapis (Items.get("Lapis"), 2),
 		Gold (Items.get("Gold Ore"), 4),
 		Gem (Items.get("Gem"), 6);
-		
+
 		private Item drop;
 		public final int color;
-		
+
 		OreType(Item drop, int color) {
 			this.drop = drop;
 			this.color = color;
 		}
-		
+
 		protected Item getOre() {
 			return drop.clone();
 		}
     }
-	
+
 	protected OreTile(OreType o) {
 		super((o == OreTile.OreType.Lapis ? "Lapis" : o.name() + " Ore"), new Sprite(24 + o.color, 0, 2, 2, 1));
         this.type = o;
@@ -67,7 +67,7 @@
 		if (item instanceof ToolItem) {
 			ToolItem tool = (ToolItem) item;
 			if (tool.type == ToolType.Pickaxe) {
-				if (player.payStamina(6 - tool.level) && tool.payDurability()) {
+				if (player.payStamina(6 - ToolItem.LEVELS.get(tool.level)) && tool.payDurability()) {
 					hurt(level, xt, yt, 1);
 					return true;
 				}
@@ -75,16 +75,16 @@
 		}
 		return false;
 	}
-	
+
     public Item getOre() {
         return type.getOre();
     }
-    
+
 	public void hurt(Level level, int x, int y, int dmg) {
 		int damage = level.getData(x, y) + 1;
 		int oreH = random.nextInt(10) + 3;
 		if (Game.isMode("Creative")) dmg = damage = oreH;
-		
+
 		level.add(new SmashParticle(x * 16, y * 16));
 		Sound.monsterHurt.play();
 

