--- a\minicraft\level\tile\PathTile.java
+++ b\minicraft\level\tile\PathTile.java
@@ -23,7 +23,7 @@
         if (item instanceof ToolItem) {
             ToolItem tool = (ToolItem) item;
             if (tool.type == ToolType.Shovel) {
-                if (player.payStamina(4 - tool.level) && tool.payDurability()) {
+                if (player.payStamina(4 - ToolItem.LEVELS.get(tool.level)) && tool.payDurability()) {
                     level.setTile(xt, yt, Tiles.get("Hole"));
                     Sound.monsterHurt.play();
                     level.dropItem(xt * 16 + 8, yt * 16 + 8, Items.get("Dirt"));

