--- a\minicraft\level\tile\RockTile.java	2021-09-08 21:34:58.000000000 +0800
+++ b\minicraft\level\tile\RockTile.java	2022-09-13 19:31:06.575240400 +0800
@@ -51,10 +51,10 @@
 	public boolean interact(Level level, int xt, int yt, Player player, Item item, Direction attackDir) {
 		if (item instanceof ToolItem) {
 			ToolItem tool = (ToolItem) item;
-			if (tool.type == ToolType.Pickaxe && player.payStamina(4 - tool.level) && tool.payDurability()) {
+			if (tool.type == ToolType.Pickaxe && player.payStamina(4 - ToolItem.LEVELS.get(tool.level)) && tool.payDurability()) {
 				// Drop coal since we use a pickaxe.
 				dropCoal = true;
-				hurt(level, xt, yt, random.nextInt(10) + (tool.level) * 5 + 10);
+				hurt(level, xt, yt, random.nextInt(10) + (ToolItem.LEVELS.get(tool.level)) * 5 + 10);
 				return true;
 			}
 		}
