--- a\minicraft\level\tile\SaplingTile.java
+++ b\minicraft\level\tile\SaplingTile.java
@@ -9,23 +9,21 @@
 
 public class SaplingTile extends Tile {
 	private static Sprite sprite = new Sprite(12, 1, 1);
-	
+
 	private Tile onType;
 	private Tile growsTo;
-	
+
 	protected SaplingTile(String name, Tile onType, Tile growsTo) {
 		super(name, sprite);
 		this.onType = onType;
 		this.growsTo = growsTo;
-		connectsToSand = onType.connectsToSand;
-		connectsToGrass = onType.connectsToGrass;
-		connectsToFluid = onType.connectsToFluid;
+		connections.addAll(onType.connections);
 		maySpawn = true;
 	}
 
 	public void render(Screen screen, Level level, int x, int y) {
 		onType.render(screen, level, x, y);
-		
+
 		sprite.render(screen, x * 16, y * 16);
 	}
 

