--- a\minicraft\level\tile\Tile.java
+++ b\minicraft\level\tile\Tile.java
@@ -21,7 +21,7 @@
 	/**
 	 * This is used by wall tiles to get what material they're made of.
 	 */
-	protected enum Material {
+	public enum Material {
 		Wood(ToolType.Axe),
 		Stone(ToolType.Pickaxe),
 		Obsidian(ToolType.Pickaxe);
@@ -37,27 +37,27 @@
 			return requiredTool;
 		}
 	}
-	
+
 	public final String name;
-	
+
 	public byte id;
-	
+
 	public boolean connectsToGrass = false;
 	public boolean connectsToSand = false;
 	public boolean connectsToFluid = false;
 	public int light;
 	protected boolean maySpawn;
-	
+
 	protected Sprite sprite;
 	protected ConnectorSprite csprite;
-	
+
 	{
 		light = 1;
 		maySpawn = false;
 		sprite = null;
 		csprite = null;
 	}
-	
+
 	protected Tile(String name, Sprite sprite) {
 		this.name = name.toUpperCase();
 		this.sprite = sprite;
@@ -67,14 +67,14 @@
 		csprite = sprite;
 	}
 
-	
+
 	/** This method is used by tiles to specify the default "data" they have in a level's data array.
 		Used for starting health, color/type of tile, etc. */
 	// At least, that was the idea at first...
 	public int getDefaultData() {
 		return 0;
 	}
-	
+
 	/** Render method, used in sub-classes */
 	public void render(Screen screen, Level level, int x, int y) {
 		if (sprite != null)
@@ -82,9 +82,9 @@
 		if (csprite != null)
 			csprite.render(screen, level, x, y);
 	}
-	
+
 	public boolean maySpawn() { return maySpawn; }
-	
+
 	/** Returns if the player can walk on it, overrides in sub-classes  */
 	public boolean mayPass(Level level, int x, int y, Entity e) {
 		return true;
@@ -115,13 +115,13 @@
 	 * @param dmg The damage taken.
 	 */
 	public void hurt(Level level, int x, int y, int dmg) {}
-	
+
 	/** What happens when you run into the tile (ex: run into a cactus) */
 	public void bumpedInto(Level level, int xt, int yt, Entity entity) {}
-	
+
 	/** Update method */
 	public boolean tick(Level level, int xt, int yt) { return false; }
-	
+
 	/** What happens when you are inside the tile (ex: lava) */
 	public void steppedOn(Level level, int xt, int yt, Entity entity) {}
 
@@ -138,10 +138,10 @@
 	public boolean interact(Level level, int xt, int yt, Player player, Item item, Direction attackDir) {
 		return false;
 	}
-	
+
 	/** Sees if the tile connects to a fluid. */
 	public boolean connectsToLiquid() { return connectsToFluid; }
-	
+
 	public int getData(String data) {
 		try {
 			return Integer.parseInt(data);
@@ -149,38 +149,38 @@
 			return 0;
 		}
 	}
-	
+
 	public boolean matches(int thisData, String tileInfo) {
 		return name.equals(tileInfo.split("_")[0]);
 	}
-	
+
 	public String getName(int data) {
 		return name;
 	}
-	
+
 	public static String getData(int depth, int x, int y) {
 		try {
 			byte lvlidx = (byte) World.lvlIdx(depth);
 			Level curLevel = World.levels[lvlidx];
 			int pos = x + curLevel.w * y;
-			
+
 			int tileid = curLevel.tiles[pos];
 			int tiledata = curLevel.data[pos];
-			
+
 			return lvlidx + ";" + pos + ";" + tileid + ";" + tiledata;
 		} catch(NullPointerException | IndexOutOfBoundsException ignored) {
 		}
-		
+
 		return "";
 	}
-	
+
 	@Override
 	public boolean equals(Object other) {
 		if (!(other instanceof Tile)) return false;
 		Tile o = (Tile) other;
 		return name.equals(o.name);
 	}
-	
+
 	@Override
 	public int hashCode() { return name.hashCode(); }
 }

