--- a\minicraft\level\tile\TorchTile.java
+++ b\minicraft\level\tile\TorchTile.java
@@ -12,14 +12,14 @@
 
 public class TorchTile extends Tile {
 	private static Sprite sprite = new Sprite(11, 3, 0);
-	
+
 	private Tile onType;
-	
+
 	public static TorchTile getTorchTile(Tile onTile) {
 		int id = onTile.id & 0xFF;
 		if(id < 128) id += 128;
 		else System.out.println("Tried to place torch on torch tile...");
-		
+
 		if(Tiles.containsTile(id))
 			return (TorchTile)Tiles.get(id);
 		else {
@@ -28,24 +28,22 @@
 			return tile;
 		}
 	}
-	
+
 	private TorchTile(Tile onType) {
 		super("Torch "+ onType.name, sprite);
 		this.onType = onType;
-		this.connectsToSand = onType.connectsToSand;
-		this.connectsToGrass = onType.connectsToGrass;
-		this.connectsToFluid = onType.connectsToFluid;
+		connections.addAll(onType.connections);
 	}
-	
+
 	public void render(Screen screen, Level level, int x, int y) {
 		onType.render(screen, level, x, y);
 		sprite.render(screen, x * 16 + 4, y * 16 + 4);
 	}
-	
+
 	public int getLightRadius(Level level, int x, int y) {
 		return 5;
 	}
-	
+
 	public boolean interact(Level level, int xt, int yt, Player player, Item item, Direction attackDir) {
 		if(item instanceof PowerGloveItem) {
 			level.setTile(xt, yt, this.onType);

