--- a\minicraft\level\tile\TreeTile.java	2021-09-08 21:34:58.000000000 +0800
+++ b\minicraft\level\tile\TreeTile.java	2022-09-13 19:32:03.798095500 +0800
@@ -84,8 +84,8 @@
 		if (item instanceof ToolItem) {
 			ToolItem tool = (ToolItem) item;
 			if (tool.type == ToolType.Axe) {
-				if (player.payStamina(4 - tool.level) && tool.payDurability()) {
-					hurt(level, xt, yt, random.nextInt(10) + (tool.level) * 5 + 10);
+				if (player.payStamina(4 - ToolItem.LEVELS.get(tool.level)) && tool.payDurability()) {
+					hurt(level, xt, yt, random.nextInt(10) + (ToolItem.LEVELS.get(tool.level)) * 5 + 10);
 					return true;
 				}
 			}
