--- a\minicraft\core\Renderer.java	2021-09-08 21:34:58.000000000 +0800
+++ b\minicraft\core\Renderer.java	2022-09-13 19:22:43.028989200 +0800
@@ -284,7 +284,7 @@
 		if (player.activeItem instanceof ToolItem) {
 			// Draws the text
 			ToolItem tool = (ToolItem) player.activeItem;
-			int dura = tool.dur * 100 / (tool.type.durability * (tool.level+1));
+			int dura = tool.dur * 100 / (tool.type.durability * (ToolItem.LEVELS.get(tool.level)+1));
 			int green = (int)(dura * 2.55f);
 			Font.drawBackground(dura + "%", screen, 164, Screen.h - 16, Color.get(1, 255 - green, green, 0));
 		}
